rackcity/views/decommission_views.py

```diff
@@ -31,6 +32,12 @@ def decommission_asset(request):
     Decommission a live asset
     """
     data = JSONParser().parse(request)
+    change_plan = None
+
```

Comment:
you need to keep the asset permission check at the top of this view - and you will need to check permission on asset_cp if there's a change plan before you do anything else

Reply:
it's below your code changes in the view, starts with...
```
if not user_has_asset_permission(
        request.user,
        asset.rack.datacenter
    ):
```

Reply:
moved the code up! do I need to additional work to check permission on asset_cp?

Reply:
i thought it was datacenter specific?

Reply:
jk i see what you mean, refactoring this method